test/units/modules/test_netapp_utils.py

```diff
@@ -0,0 +1,89 @@
+# Copyright (c) 2018 NetApp
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+''' unit tests for module_utils netapp.py '''
+from __future__ import absolute_import, division, print_function
+import sys
+import inspect
+import os
+
+try:
+    import netapp_lib.api.zapi
+    sys.path.insert(0, os.path.dirname(inspect.getfile(netapp_lib.api.zapi)))
+    from netapp_lib.api.zapi import zapi
+    HAS_NETAPP_ZAPI = True
+except ImportError:
+    HAS_NETAPP_ZAPI = False
+
+try:
+    from ansible.module_utils.ansible_release import __version__ as ansible_version
+except ImportError:
+    ansible_version = 'unknown'
+
+from units.compat import unittest
+import ansible.module_utils.netapp as my_module
+HAS_NETAPP_ZAPI_MSG = "pip install netapp_lib is required"
+
+
+class MockONTAPConnection(object):
+    ''' mock a server connection to ONTAP host '''
+
+    def __init__(self, kind=None, parm1=None):
+        ''' save arguments '''
+        self.type = kind
+        self.parm1 = parm1
+        self.xml_in = None
+        self.xml_out = None
+
+    def invoke_successfully(self, xml, enable_tunneling):  # pylint: disable=unused-argument
+        ''' mock invoke_successfully returning xml data '''
+        self.xml_in = xml
+        if self.type == 'vserver':
+            xml = self.build_vserver_info(self.parm1)
+        self.xml_out = xml
+        return xml
+
+    @staticmethod
+    def build_vserver_info(vserver):
+        ''' build xml data for vserser-info '''
+        xml = zapi.NaElement('xml')
+        attributes = zapi.NaElement('attributes-list')
+        attributes.add_node_with_children('vserver-info',
+                                          **{'vserver-name': vserver})
+        xml.add_child_elem(attributes)
+        return xml
+
+
+class TestEMSLogVersion(unittest.TestCase):
```

Comment:
New unit tests should be written using pytest style instead of unittest. The biggest difference being the use of functions and fixtures instead of classes.